### DISCUSSION TITLE
Scenario based Modelling for my custom country

### DISCUSSION THREAD

**Discussion by Constable1819** [pypsa-meets-earth/pypsa-earth#1269] [General]
Hi....Is scenario based analysis possible with pypsa earth? I am trying to determine the optimal energy mix for a carbon free electricity scenario (self sufficiency) in 2050 for Namibia. How do I do these edits in the config file?

**Comment by ekatef** [comment]
Hello @Constable1819, great to hear that you have managed to setup the model!

The configuration parameters are described in [Configuration](https://pypsa-earth.readthedocs.io/en/latest/configuration.html) section of the documentation.
Given your interest to the net-zero scenario, `Co2L` is relevant for you, while other settings depend on your problem definition.

**Reply by Constable1819** [reply]
Halo @ekatef I did go through Configuration. My question is, is it possible for the model to produce capacity values for renewables when for example I want self sufficiency in 2050, with net zero? when i set Co2 under electricity to 0 (under the assumption that this will mean that all electricity is produced from renewables), it still shows that Co2 limit is 7.75e+7.

**Reply by ekatef** [reply]
> Halo @ekatef I did go through Configuration. My question is, is it possible for the model to produce capacity values for renewables when for example I want self sufficiency in 2050, with net zero? when i set Co2 under electricity to 0 (under the assumption that this will mean that all electricity is produced from renewables), it still shows that Co2 limit is 7.75e+7.

I confirm that your understand the logic properly 🙂 Great work!

Just to be sure we are talking about same things: you mean modifying `co2limit` in your `config.yaml`, right? It should work which you can check by looking into `n.global_constraint` in a pre-solved and solved networks.

Another option is using `Co2L` trigger in `{opts}` [wildcards](https://pypsa-earth.readthedocs.io/en/latest/wildcards.html) and `automatic_emission: true`.

To give a bit of context, this code fragment sets CO2 constraints using one or the options (`co2limit` or `Co2` wildcard) :
https://github.com/pypsa-meets-earth/pypsa-earth/blob/ef525ee237b27916caf9a94a2d2157d25bd962f3/scripts/prepare_network.py#L375-L381

**Reply by Constable1819** [reply]
I will try this. In the config file, there are a lot of things I don't need/use. I just want to model electricity. Can I delete most of them or it will affect the network? Like the sector options section

**Reply by ekatef** [reply]
Under the current design, it's more reliable to have all the parameters in the `config.yaml` as it gives your more control on the simulation parameters. If you remove some from `config.yaml` they will be taken from `config.default.yaml` which can lead to some unexpected effects.

I do absolutely agree with you however that the current management of the configuration can definitely be improved. We are working on that, and thank your for sharing the idea!

**Comment by Constable1819** [comment]
@ekatef I got the co2 limit to be 0. However, oil and coal powerplants are still being built.
![EM capacities](https://github.com/user-attachments/assets/a4f8c710-65ff-47b5-88ff-8450473ec17b)
I tried changing my scenarios: 

scenario:
  simpl: [""]
  ll: ["v1.25"]
  clusters: [6]
  opts: [Co2L-2H-EQ1c-RES1]
  planning_horizons: # investment years for myopic and perfect; or costs year for overnight
  - 2030
  sopts:
  - "144H"
  demand:
  - "AB"

  opts: [Co2L-2H-EQ1c-RES1] 
I want my country to produce 100% of its electricity and 100% share renewables, basically net zero. Also included line expansion of 25% v1.25.  it still remained the same as shown below
![image](https://github.com/user-attachments/assets/bdf814bf-973a-4389-930b-d94955bcf905)

**Reply by ekatef** [reply]
The generators are not being removed from the system if they have existed before, while `n.generators.p_nom` shows the installed capacity of "today". The relevant parameters to check if the CO2 constraint is working is `n.generators_t`.

Regarding the constraints, not sure about `EQ1c` and `RES1` constraints as they may feel be redundant. Some additional testing could be recommended to check that everything is working as expected

**Reply by Constable1819** [reply]
i used n.generators_t.p_max_pu to check this before but because i was still stuck on p_nom, i thought something was wrong and all were below zero so i assumed they are capacity factors. Thank you @ekatef . 

Still on the same question, when you say today, you mean currently right? So then how do I show the required renewable energy to satisfy current total demand for my time period in my modelled year which is 2050?
 
I would also like to know land requirements for these capacities, is it possible?

**Reply by ekatef** [reply]
Pleasure to help 🙂 

Yeah, I meant with "today" a state of the power system which is as close as possible to it's real state right now which implies the installed capacity and the power grid.

The simplest approach to estimate the land requirements would be to extract the overall installed capacity for each type of renewable power (`n.generators.p_nom_opt`) and multiply by an average land usage per MW. Actually, we account for quite many details when determining land availability for constructing the renewable power. [This notebook ](https://github.com/pypsa-meets-earth/documentation/blob/main/notebooks/landuse-availability.ipynb)gives some hints on that. But I'm not sure if it wouldn't be an over-complication for to answer your question
